pkg/urls/well_known.go

```diff
@@ -18,5 +18,54 @@ limitations under the License.
 
 package urls
 
+import (
+	"github.com/openshift-online/ocm-cli/pkg/config"
+	sdk "github.com/openshift-online/ocm-sdk-go"
+)
+
 // OfflineTokenPage is the URL of the page used to generate offline access tokens.
 const OfflineTokenPage = "https://console.redhat.com/openshift/token" // #nosec G101
+
+const (
+	OCMProductionURL  = "https://api.openshift.com"
+	OCMStagingURL     = "https://api.stage.openshift.com"
+	OCMIntegrationURL = "https://api.integration.openshift.com"
+)
+
+var OCMURLAliases = map[string]string{
+	"production":  OCMProductionURL,
+	"prod":        OCMProductionURL,
+	"prd":         OCMProductionURL,
+	"staging":     OCMStagingURL,
+	"stage":       OCMStagingURL,
+	"stg":         OCMStagingURL,
+	"integration": OCMIntegrationURL,
+	"int":         OCMIntegrationURL,
+}
+
+func ValidOCMUrlAliases() []string {
+	keys := make([]string, 0, len(OCMURLAliases))
+	for k := range OCMURLAliases {
+		keys = append(keys, k)
+	}
+	return keys
+}
+
+// URL Precedent (from highest priority to lowest priority):
+//  1. runtime `--url` cli arg (key found in `urlAliases`)
+//  2. runtime `--url` cli arg (non-empty string)
+//  3. config file `URL` value (non-empty string)
+//  4. sdk.DefaultURL
+func ResolveGatewayURL(optionalParsedCliFlagValue string, optionalParsedConfig *config.Config) string {
```

Comment:
How do we handle a scenario where:
User wants to log in to prod : `ocm login --url=Prod`
But this would fail since Prod is not a valid alias.
We have a stage URL stored in config, so we log the user into OCM stage.
Should we show any proper messages in this case?

Reply:
I dont follow, "production", "prod" and "prd" are all valid aliases. If you mean how do we detect a mistaken prod alias vs a raw url and give a better error message, we dont do that today. I can add that, though.

Reply:
So its tricky to know the user's exact intent. Were they trying to use an alias and typoed it or used one that doesnt exist, or did they input an invalid url. Which is probably why they didnt bother before, they just pass the this-is-not-a-known-alias `--url` value to the underlying SDK and let it throw up an error if it wasnt actually a good url. 

So I made ResolveGatewayURL, as a final step, try to parse the resolved URL using url.ParseRequestURI and return an helpful error message if that fails.

```
> ocm login --token=$OCM_OFFLINE_ACCESS_TOKEN --url="Prd"      
Error: parse "Prd": invalid URI for request

URL Source: flag
Expected an absolute URI or an absolute path (e.g. https://api.openshift.com) or a case-sensitive alias, one of: [int, production, prod, prd, staging, stage, stg, integration]
```

Logic interpreted that as the user trying to input an invalid url because its not a known alias, but we display our expectations, including the known aliases and the fact that they are case sensitive in the error message to help the user know what to do about it. As a bonus, this also works if their saved `cfg.URL` gets mangled for some reason.

Reply:
Also to be clear, even before i improved things `ocm login --url=Prod` would have failed, not fall back to the stored config, just like it does on main today. It would have just given you a slightly less helpful error message